stan/math/opencl/prim/categorical_logit_glm_lpmf.hpp

```diff
@@ -71,21 +71,19 @@ return_type_t<T_alpha, T_beta> categorical_logit_glm_lpmf(
     return 0;
   }
 
-  T_alpha_ref alpha_ref = alpha;
-  T_beta_ref beta_ref = beta;
+  const auto& y_val = eval(value_of(y));
+  const auto& x_val = eval(value_of(x));
+  const auto& alpha_val = eval(value_of(alpha));
+  const auto& beta_val = eval(value_of(beta));
 
-  const auto& alpha_val = value_of_rec(alpha_ref);
-  const auto& beta_val = value_of_rec(beta_ref);
+  const auto& y_val_cl = to_matrix_cl(y_val);
+
+  matrix_cl<double> x_beta_cl = x_val * beta_val;
```

Comment:
Does this have to  explicitly state that its matrix_cl<double>? Just double checking that it cant be anything else.

Reply:
I think so. `x` and `beta` must be one of `var_value<matrix_cl>`, `matrix_cl` or kernel generator expression. In lines 74-77 we are extracting values from any `var_value` inputs and evaluating any expressions. So this can only be `matrix_cl`.